napari/_vispy/_tests/test_vispy_multiscale.py

```diff
@@ -156,6 +156,39 @@ def test_image_screenshot_zoomed(make_napari_viewer):
     )
 
 
+@skip_on_win_ci
+@skip_local_popups
+def test_multiscale_zoomed_out(make_napari_viewer):
```

Comment:
@andy-sweet maybe I'm confused, but how sure are you that this test fails on main? By my understanding, the current behaviour is that the *canvas size* determines the minimum resolution level. This means that for a canvas size of 1600x1600, you expect level 1 to be loaded, even in main.

The problem is that in main, it doesn't matter *how much* you zoom out, the viewer will never go to level 2, even if the whole dataset only takes up one pixel on screen. This is dumb. 😂 

So, my suggestion is that you make the camera rect be *much* bigger than the data coords, say 34000x34000, and make sure that level *2* is loaded.

What do you think?

Reply:
This does fail on main right now. The canvas size does influence which level is chosen, but it's not the only factor. For example, if we change the test case here to use `camera_rect_size = 800`, then `assert layer.data_level == 0` passes on main - i.e. being zoomed in works, but being zoomed out doesn't because in that case we also clip the canvas corners in the layer's data coordinates to the layer's extent.

The tests in `test_multiscale` cover more cases and sweep a few expected levels. The goal of this test was to have at least one slower/integrated test that covers the logic in `QtViewer.on_draw` rather than just the faster/headless stuff in `Layer._update_draw`. As there's only one test here, I chose a just-right/goldilocks value because when there's only one test I find those catch more errors than when testing at one extreme.

But, I just noticed we have `test_multiscale_screenshot_zoomed` (which tests being zoomed in), so I'm fine with changing this to the extreme case you suggested. If you like, we could also parameterize this test to cover all three levels (that's what I initially did), but I think it's probably not worth the extra test runtime.